library/std/src/os/uefi/env.rs

```diff
@@ -0,0 +1,129 @@
+//! UEFI-specific extensions to the primitives in `std::env` module
+
+use super::raw::{BootServices, RuntimeServices, SystemTable};
+use crate::ffi::c_void;
+use crate::io;
+use crate::mem::MaybeUninit;
+use crate::ptr::NonNull;
+use crate::sync::atomic::{AtomicPtr, Ordering};
+use r_efi::efi::{Guid, Handle};
+use r_efi::system;
+
+static GLOBAL_SYSTEM_TABLE: AtomicPtr<SystemTable> = AtomicPtr::new(crate::ptr::null_mut());
+static GLOBAL_SYSTEM_HANDLE: AtomicPtr<c_void> = AtomicPtr::new(crate::ptr::null_mut());
+
+/// Initializes Global Atomic Pointers to SystemTable and Handle.
+/// Should only be called once in the program execution under normal circumstances.
+/// The caller should ensure that the pointers are valid.
+pub(crate) fn init_globals(handle: NonNull<c_void>, system_table: NonNull<SystemTable>) {
+    GLOBAL_SYSTEM_TABLE.store(system_table.as_ptr(), Ordering::SeqCst);
+    GLOBAL_SYSTEM_HANDLE.store(handle.as_ptr(), Ordering::SeqCst);
+}
+
+#[unstable(feature = "uefi_std", issue = "none")]
+/// Get the SystemTable Pointer.
```

Comment:
Please put attributes after the doc comment. This is nicer especially for longer doc comments.